### PR TITLE
Adapt package structure

### PR DESCRIPTION
**Pull request by schlessera**
This PR makes changes to the package structure to fit the one from the bundled plugins:
- Use `WP_CLI\Doctor` as the namespace root
- Use PSR-4 autoloader
- Use latest `wp-cli-tests` framework with Behat v3

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]

**Comment by swissspidy** [comment]
@schlessera @danielbachhuber Apologies for the noise here, but I thought I'd pick this up and help fix all the failing tests. This is now ready for review!

I don't mind the namespace change, I think it should be fine. Just noting that this handbook page needs updating after merge: https://github.com/wp-cli/handbook/blob/7c792d2a11448dab77f23d8055f82553e6dff2a6/doctor-customize-config.md

File: doctor.yml [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
+  check: ThemeUpdate
 cache-flush:
-  check: Cache_Flush
+  check: CacheFlush
 php-in-upload:
-  check: PHP_In_Upload
+  check: PHPInUpload
 language-update:
-  check: Language_Update
+  check: LanguageUpdate
```
Comment (danielbachhuber) [inline doctor.yml:44]: This will break folks existing `doctor.yml` files.

**Review by danielbachhuber — COMMENTED** [review]
I'm not really a fan of this pull request as it currently is. It will cause a lot of breakage for existing users, for very little real-world benefit.

If we're committed to renaming the classes, then I think we should add backcompat for the existing classes (and make sure there's backcompat for the `doctor.yml` file).

File: doctor.yml (hunk 1 above)
Comment (swissspidy) [reply doctor.yml:44]: We could keep the class name but just change the namespace

**Comment by swissspidy** [comment]
Makes sense.

Btw, any idea why this one test is failing on 6.2?
The 3.7 one I haven‘t looked into yet

**Comment by danielbachhuber** [comment]
> Btw, any idea why this one test is failing on 6.2?

```
$ wp core version
6.2
$ wp core check-update
+---------+-------------+-----------------------------------------------------------------------+
| version | update_type | package_url                                                           |
+---------+-------------+-----------------------------------------------------------------------+
| 6.2.3   | minor       | https://downloads.wordpress.org/release/wordpress-6.2.3-partial-0.zip |
| 6.3.2   | major       | https://downloads.wordpress.org/release/wordpress-6.3.2.zip           |
+---------+-------------+-----------------------------------------------------------------------+
$ wp doctor check core-update
+-------------+--------+----------------------------------------------------------------------+
| name        | status | message                                                              |
+-------------+--------+----------------------------------------------------------------------+
| core-update | error  | Updating to WordPress' newest minor version is strongly recommended. |
+-------------+--------+----------------------------------------------------------------------+
Error: 1 check reports 'error'.
```

It looks like `main` is passing because `behat.yml` isn't detected and no tests are being run.

**Comment by swissspidy** [comment]
Yes, that I realized. That‘s why I did all these commits to fix tests. Just thought it was weird that it failed for that particular job. But I can adapt the test 👍

**Comment by swissspidy** [comment]
OK looks like `@require-wp-latest` doesn't seem to actually work 🤔

**Comment by danielbachhuber** [comment]
> OK looks like `@require-wp-latest` doesn't seem to actually work 🤔

@swissspidy Lovely. Maybe it's not actually a thing!

For `Scenario: WordPress is up to date`, you could run `wp core update` prior to `wp doctor check core-update`. I'd probably do the same for `Scenario: Use --spotlight to focus on warnings and errors`.

**Comment by swissspidy** [comment]
Thanks, yeah I‘ll try that again. 

Amd then I guess I‘ll review all existing usage of `@require-wp-latest` in the code base 🤔

Any idea on the WP 3.7 test? Can‘t run them locally as I don‘t have old PHP installed.

**Comment by danielbachhuber** [comment]
> Any idea on the WP 3.7 test? Can‘t run them locally as I don‘t have old PHP installed.

Nothing immediately obvious, no.

**Comment by swissspidy** [comment]
@ernilambar Since you worked on this repo recently, maybe this PR is something you'd be interested in bringing over the finish line?

**Comment by ernilambar** [comment]
> @ernilambar[](https://profiles.wordpress.org/rabmalin) Since you worked on this repo recently, maybe this PR is something you'd be interested in bringing over the finish line?

Sure. I will work on it.

**Comment by ernilambar** [comment]
Closing in favor of https://github.com/wp-cli/doctor-command/pull/194

**Pull request closed without merging**